packages/massa-web3/src/experimental/smartContract.ts

```diff
@@ -207,13 +239,29 @@ export class SmartContract {
   }
 
   /**
-   * Estimates the gas required for an operation.
-   * Currently, it returns a predefined maximum gas value.
-   * @returns A promise that resolves to the estimated gas amount in bigint.
-   * TODO: Implement dynamic gas estimation using dry run call.
-   * TODO: Remove static if needed.
+   * Returns the gas estimation for a given function.
+   *
+   * @param func - The function to estimate the gas cost.
+   * @returns The gas estimation for the function.
    */
-  static async getGasEstimation(): Promise<bigint> {
-    return MAX_GAS_CALL
+  async getGasEstimation(
+    func: string,
+    parameter: Uint8Array,
+    callerAddress: Address,
```

Comment:
Should be an option IMHO

Reply:
If no callerAddress the read operation fail

Reply:
But the read function has it as an optional parameter.